test/python/taskbasedsimulator/benchmark.py

```diff
@@ -0,0 +1,346 @@
+            number_of_newgates.append((j, testqc_transpile.size()))
+        elif j == 'mcrx':
+            testqc = QuantumCircuit(2, 2)
+            testqc.crx(60, 0, 1)
+            testqc_transpile = transpile(testqc, basis_gates=basis_gates_elementary, optimization_level=0)
+            number_of_newgates.append((j, testqc_transpile.size()))
+        elif j == 'cry':
+            testqc = QuantumCircuit(2, 2)
+            testqc.cry(60, 0, 1)
+            testqc_transpile = transpile(testqc, basis_gates=basis_gates_elementary, optimization_level=0)
+            number_of_newgates.append((j, testqc_transpile.size()))
+        elif j == 'mcry':
+            testqc = QuantumCircuit(2, 2)
+            testqc.cry(60, 0, 1)
+            testqc_transpile = transpile(testqc, basis_gates=basis_gates_elementary, optimization_level=0)
+            number_of_newgates.append((j, testqc_transpile.size()))
+        elif j == 'crz':
+            testqc = QuantumCircuit(2, 2)
+            testqc.crz(60, 0, 1)
+            testqc_transpile = transpile(testqc, basis_gates=basis_gates_elementary, optimization_level=0)
+            number_of_newgates.append((j, testqc_transpile.size()))
+        elif j == 'mcrz':
+            testqc = QuantumCircuit(2, 2)
+            testqc.crz(60, 0, 1)
+            testqc_transpile = transpile(testqc, basis_gates=basis_gates_elementary, optimization_level=0)
+            number_of_newgates.append((j, testqc_transpile.size()))
+        elif j == 'cp':
+            testqc = QuantumCircuit(2, 2)
+            testqc.cp(60, 0, 1)
+            testqc_transpile = transpile(testqc, basis_gates=basis_gates_elementary, optimization_level=0)
+            number_of_newgates.append((j, testqc_transpile.size()))
+        elif j == 'cu1':
+            testqc = QuantumCircuit(2, 2)
+            testqc.cp(60, 0, 1)
+            testqc_transpile = transpile(testqc, basis_gates=basis_gates_elementary, optimization_level=0)
+            number_of_newgates.append((j, testqc_transpile.size()))
+        elif j == 'mcphase':
+            testqc = QuantumCircuit(4, 4)
+            testqc.mcp(20, [0, 1, 2], 3)
+            testqc_transpile = transpile(testqc, basis_gates=basis_gates_elementary, optimization_level=0)
+            number_of_newgates.append((j, testqc_transpile.size()))
+        elif j == 'csx':
+            testqc = QuantumCircuit(2, 2)
+            testqc.csx(0, 1)
+            testqc_transpile = transpile(testqc, basis_gates=basis_gates_elementary, optimization_level=0)
+            number_of_newgates.append((j, testqc_transpile.size()))
+        elif j == 'sxdg':
+            testqc = QuantumCircuit(2, 2)
+            testqc.sxdg(1)
+            testqc_transpile = transpile(testqc, basis_gates=basis_gates_elementary, optimization_level=0)
+            number_of_newgates.append((j, testqc_transpile.size()))
+        else:
+            number_of_newgates.append((j, 1))
+    return number_of_newgates
```

Comment:
This file still needs lots of cleanup of everything related to avg-case mode and such.
It should basically represent the evaluation in the paper.